File: dart/indice.dart
/*
Calculadora de indice de masa corporal
if...else if...else
*/

import 'utilidades.dart' as util;
import 'dart:math' as mth;

void main() {
  //declarar variables
  double peso;
  double altura;
  double imc;

  util.encabezado('Calculadora de indice de masa corporal', 150);

  peso = util.leerNumeroDecimal('Ingresa tu peso en kg');
  altura = util.leerNumeroDecimal('Ingresa tu estatura en mts');
  imc = calculoIMC(peso, altura);

  util.linea(50);
  print('El indice de masa corporal es: ${imc.toStringAsFixed(2)} - ${categoriaImc(imc).toUpperCase()}');
  util.linea(50);
}

double calculoIMC(double peso, double altura) {
  return (peso / mth.pow(altura, 2));
}

String categoriaImc(double imc) {
  if (imc < 18.5) {
    return 'Estas bajo peso';
  } else if (imc >= 18.5 && imc < 24.9) {
    return 'Estas peso normal';
  } else if (imc >= 24.9 && imc < 29.9) {
    return 'Estas sobrepeso';
  } else {
    return 'Estas en obesidad';
  }
}

File: JS/src/indice.js
/* 
calculadora de indice 
*/
import * as util from  "./utilidades.js"

const main = () =>{
    //definir variables
    let peso, altura, imc = 0;

    util.encabezado("calculadora de indice de masa corporal", 50)
    peso = util.leerNumero("ingresa tu peso en kg")
    altura = util.leerNumero("ingresa tu altura en mts")
    imc = calculoImc(peso, altura)
    util.linea(50)
    console.log(`el indice de masa corporal es de ${imc.toFixed(2)} (${categoriasImc(imc).toUpperCase()})`);
    util.linea(50)


}
const calculoImc = (peso = 0, altura = 0) => (peso / Math.pow(altura, 2)) 

const categoriasImc = function(imc = 0){
    if (imc < 18.5) {
        return 'bajo peso'
    } else if (imc >= 18.5 && imc < 24.9) {
        return 'peso normal'
    } else if (imc >= 24.9 && imc < 29.9){
        return 'sobrepeso'
    }else{
        return 'obesidad'
    }

}

main()
